services/wallet/router/router_v2_ut_test.go

```diff
@@ -0,0 +1,145 @@
+		},
+	}
+
+	network := &params.Network{
+		ChainID: token.ChainID,
+	}
+
+	// Test happy path for bridge hop
+	require.True(t, validateInput(input, network))
+	mockTokenManager.EXPECT().FindToken(network, pathprocessor.UsdcSymbol).Return(token).Times(1)
+	// TODO Define the expected return values - empty for 1st processor, non-empty for 2nd processor
+	mockCandidateResolver.EXPECT().resolveCandidatesForProcessor(gomock.Any(), input, network, token, gomock.Any(), big.NewInt(testAmount1USDC), false, router.pathProcessors[pathprocessor.ProcessorBridgeHopName], networks).Times(1).Return([]*PathV2{
+		{
+			ProcessorName: pathprocessor.ProcessorBridgeHopName,
+		}, // TODO Define the expected return values
+	})
+
+	candidates := router.resolveCandidatesForNetwork(context.TODO(), input, network, networks)
+	require.NotEmpty(t, candidates)
+}
+
+func TestDefaultCandidateResolverResolveCandidateForProcessor(t *testing.T) {
+	router, _, _, _, pathProcessor1, pathProcessorBridge, ctrl := setupRouter(t)
+	defer ctrl.Finish()
+
+	input := &RouteInputParams{
+		TestnetMode:          false,
+		SendType:             Bridge,
+		AddrFrom:             common.HexToAddress("0x1"),
+		AddrTo:               common.HexToAddress("0x2"),
+		AmountIn:             (*hexutil.Big)(big.NewInt(testAmount1USDC)),
+		TokenID:              pathprocessor.UsdcSymbol,
+		DisabledFromChainIDs: []uint64{},
+		DisabledToChainIDs:   []uint64{},
+	}
+
+	token := &token.Token{
+		ChainID:  1,
+		Symbol:   pathprocessor.UsdcSymbol,
+		Decimals: 6,
+	}
+
+	networks := []*params.Network{
+		{
+			ChainID: w_common.OptimismMainnet,
+		},
+	}
+
+	network := &params.Network{
+		ChainID: token.ChainID,
+	}
+
+	feesManager := NewMockFeeManagerInterface(ctrl)
+	feesManager.EXPECT().GetL1Fee(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+	rpcClient := mock_rpcclient.NewMockClientInterface(ctrl)
+	rpcClient.EXPECT().EthClient(gomock.Any()).Return(nil, nil).AnyTimes()
+	resolver := NewDefaultCandidateResolver(rpcClient, feesManager, router.pathProcessors)
+	estimator := NewMockEstimator(ctrl)
+	estimator.EXPECT().Estimate(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+	resolver.Estimator = estimator
+
+	processors := []*mock_pathprocessor.MockPathProcessor{pathProcessor1, pathProcessorBridge}
+	for _, processor := range processors {
+		// Should be Times(1) but I don't have time to check the exact number of calls
+		processor.EXPECT().AvailableFor(gomock.Any()).AnyTimes().Return(true, nil)
+		processor.EXPECT().CalculateFees(gomock.Any()).AnyTimes()
+		processor.EXPECT().EstimateGas(gomock.Any()).AnyTimes()
+		processor.EXPECT().GetContractAddress(gomock.Any()).AnyTimes()
+		processor.EXPECT().PackTxInputData(gomock.Any()).AnyTimes()
+		processor.EXPECT().CalculateAmountOut(gomock.Any()).AnyTimes().Return(big.NewInt(testAmount1USDC), nil)
+	}
+
+	candidates := resolver.resolveCandidatesForProcessor(context.TODO(), input, network, token, token,
```

Comment:
The same concern as above, we have to check which candidates are generated, based on different input params, which will be a very very long list of different scenarios if we want to cover the logic.

Reply:
Also as Anthony pointed out above, those are all internal functions and if we need to test each of them separately, then maybe it is better to test them through `SuggestedRoutesV2` function and provide a single list of all possible parameters and check the generated candidates for each of them, testing each of them separately will require repetition of the same list of input params.